jobs/mongodb_migration/src/mongodb_migration/migrations/_20230320163700_cache_first_rows_from_streaming.py

```diff
@@ -0,0 +1,38 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 The HuggingFace Authors.
+
+import logging
+
+from libcommon.simple_cache import CachedResponse
+from mongoengine.connection import get_db
+
+from mongodb_migration.check import check_documents
+from mongodb_migration.migration import Migration
+
+first_rows = "/first-rows"
+first_rows_from_streaming = "first-rows-from-streaming"
+first_rows_tmp = "/first-rows-TMP"
+db_name = "cache"
+
+
+# connection already occurred in the main.py (caveat: we use globals)
+class MigrationCacheUpdateFirstRows(Migration):
+    def up(self) -> None:
+        logging.info(f"Rename cache_kind field from {first_rows} to {first_rows_from_streaming}")
+        db = get_db(db_name)
+        # update existing documents with the new kind (if any) to avoid duplicates (will be deleted later)
+        db["cachedResponsesBlue"].update_many({"kind": first_rows_from_streaming}, {"$set": {"kind": first_rows_tmp}})
```

Comment:
I understand, but I'm afraid: what would occur if, for some reason, this job migration is run again: we would loose all the cache entries. Is there a way to do the same without deleting the existing entries (that must be correct, by the way). Maybe there is a way to try to update, and ignore the duplicate errors?

Reply:
I did the rollback of those lines. Sadly I could find a way to skip in update error. I think the problem with duplicated records might be caused in splits because of granularity level kind-dataset-config-split, since split-names-from-streaming  always has split as null/None. But that is just a suspect.
I would like to try the simple update_many approach with a sample prod data. If needed we could try to add a transaction before the update but not sure, would like to reproduce the issue locally first. wdyt?
Same for queue but since unicity_id is not a unique constraint, it shouldn't fail.

Reply:
By the way, I don't see how we could have existing entries with `split-first-rows-from-streaming` kind. So applying

https://github.com/huggingface/datasets-server/pull/967/files#diff-17d3c24000f1960066c66c311c9c598dde0c10fc79b2e6493830a1cafd10e2adR24-R26

should work.